front/components/ideaDetail/IdeaShow.vue

```diff
@@ -0,0 +1,288 @@
+  <v-row>
+    <v-col cols="12" :md="preview ? '' : 8" class="idea-part">
+      <div>
+        <v-row class="idea-part__header" no-gutters>
+          <v-col class="idea-part__header__title">
+            <h2 class="idea-part__header__title__label">
+              {{ idea.title }}
+            </h2>
+          </v-col>
+        </v-row>
+        <div class="idea-part__info pt-2 pb-2">
+          <v-row no-gutters>
+            <span class="idea-part__info__author">
+              <router-link
+                class="idea-part__info__author__link d-flex align-center"
+                :to="{
+                  name: 'profile-userSlug',
+                  params: { userSlug: idea.authorSlug }
+                }"
+              >
+                <img
+                  class="idea-part__author-avatar"
+                  :src="idea.authorAvatar"
+                />
+                <span class="ml-2">{{ idea.authorName }}</span>
+              </router-link>
+            </span>
+            <span class="muted ml-2 idea-date">{{
+              idea.createdDate | toRelativeDate
+            }}</span>
+          </v-row>
+        </div>
+        <v-row cols="auto" offset="1">
+          <menu-panel
+            :editable="idea.userId === userId"
+            :preview="preview"
+            :idea="idea"
+            @enable-edit-mode="$emit('edit')"
+            @saved-state-changed="onIdeaSaveStateChanged"
+            @liked-state-changed="onIdeaLikeStateChanged"
+            @on-notification="onNotification"
+            @on-idea-shared="onIdeaShared"
+            @on-delete-idea="onDeleteIdea"
+            @on-idea-visibility-changed="onIdeaVisibilityChanged"
+            @on-idea-visibility-change-error="onIdeaVisibilityChangeError"
+            @comments-btn-click="commentsBtnClick"
+            @click.native.stop
+          ></menu-panel>
+        </v-row>
+      </div>
+      <div class="idea-part__content-container">
+        <div class="idea-part__content">
+          <div class="idea-part__idea-content hide-scrollbar fade-bottom">
+            <idea-content
+              :collapsed="!expanded"
+              :content="ideaContent"
+            ></idea-content>
+            <div
+              v-if="preview"
+              class="text-center cursor-pointer"
+              @click.stop="$emit('view-preview', idea)"
+            >
+              <span class="link-highlight">View Idea</span>
+            </div>
+          </div>
+        </div>
+        <div v-if="!preview" class="idea-part__tags-panel">
+          <div class="tagsContainer">
+            <v-chip v-for="(item, index) in ideaTags" :key="index" class="tag"
+              >{{ item }}
+            </v-chip>
+          </div>
+        </div>
+      </div>
+    </v-col>
+    <v-col v-if="!preview" cols="12" md="4" class="comments-section">
+      <idea-comments
+        v-if="expanded"
+        ref="ideaComments"
+        :idea="idea"
+        @onNotification="onNotification"
```

Comment:
events better to be dash cased per vue guide https://vuejs.org/v2/guide/components-custom-events.html#Event-Names

Reply:
That was introduced before me, and will be removed in https://github.com/ezl/dailyidea.com/issues/627 this notifications should be global.